### PR TITLE
Money request header scrollable

### PR DESCRIPTION
**Pull request by ShogunFire**
### Details

We are dividing the header in two part and one part is going to be in the list, we put it as the footer even if we want it to act like a header because the list in inverted. We are doing this change because the header was taking too much space on little screens. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/19389
PROPOSAL: https://github.com/Expensify/App/issues/19389#issuecomment-1582516671

### Tests

1. Go to staging dot on mweb chrome
2. Click on + icon
3. Select request money > enter amount
4. Select a contact and submit
6. Click on the payment request in the new DM
7. This takes you to a new screen
8. Verify that the header composed of:
- Mail 1
- To
- Mail 2
- Amount
- Description
- Date
is shown at the top
9. Open the keyboard and write some messages until the messages reach the header
10. Verify that only Mail1 stays fix and the rest of the header become scrollable 

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Turn your connection off
2. Do the same steps as above
3. Verify that everything that needs to be grey is grey

### QA Steps

1. Go to staging dot on mweb chrome
2. Click on + icon
3. Select request money > enter amount
4. Select a contact and submit
6. Click on the payment request in the new DM
7. This takes you to a new screen
8. Verify that the header composed of:
- Mail 1
- To
- Mail 2
- Amount
- Description
- Date
is shown at the top
9. Open the keyboard and write some messages until the messages reach the header
10. Verify that only Mail1 stays fix and the rest of the header become scrollable 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/19537677/9ae477e4-c284-4249-b32c-980bfc9f16a5

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/19537677/23d86662-368b-4ab9-b0a3-dfe3fde7b3fc

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/19537677/88ff0bba-035a-4702-b81c-71140e33a708

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/19537677/e250e013-1a7b-46bc-bdef-cbb75d023469

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/19537677/1d489ab3-c5d8-4be5-a2e2-40e77558f217

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/19537677/90f0a12d-ac07-4949-97c3-970ceba72f0a

</details>

### CODE REVIEW

**Comment by ShogunFire** [comment]
Hello @aimane-chnaif, it looks like there is a lint problem unrelated to my code, I ran npm run lint on my computer and fixed some errors even if I think there are still some syntax errors that they don't tell me, I also changed the name of the files as MaciejSWM suggested. 

The biggest change from the initial branch is the android version of InvertedList, I did it because Lint asked me to validate the props, I copied the way it was done in index.js I hope it's good. 

Maybe you can start the review until they fix the lint error, thanks

File: .github/actionlint.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # See https://github.com/rhysd/actionlint/blob/main/docs/config.md
 self-hosted-runner:
   labels:
-    - ubuntu-latest-xl
+    - ubuntu-20.04-64core
```
Comment (AndrewGable) [inline .github/actionlint.yaml:4]: Please revert this change

**Review by AndrewGable — DISMISSED** [review]

**Comment by AndrewGable** [comment]
Looks like lint is passing now 👍 

Screenshot 2023-06-30 at 12 59 03 PM

**Review by AndrewGable — COMMENTED** [review]
Looks like lint is passing now 👍 

Screenshot 2023-06-30 at 12 59 03 PM

**Comment by ShogunFire** [comment]
Yes... Weird... Thank you @AndrewGable

**Comment by ShogunFire** [comment]
Hello @aimane-chnaif I have trouble degugging on android so I don't see the error in the console but I made a change on prop type. Can you check if it works now or can you send me the error ?

I also merged.

I have a weird bug on android, when I open the money request, sometimes it doesn't show the list, maybe it's related to this error that you have:

https://github.com/Expensify/App/assets/19537677/e9da63e7-df1e-4562-a6b5-e5637c759112

**Comment by ShogunFire** [comment]
I changed it thanks, I just tested on main and it's also happening

One last thing, I don't really understand the code in InvertedList android and maybe there is a way to simplify like this, it seems to work and it is hat we have in the normal (not inverted) FlatList for android: 

image

What do you think ?

**Review by aimane-chnaif — APPROVED** [review]
LGTM 🎉 

@madmax330 @shawnborton all yours!

**Comment by shawnborton** [comment]
Hmm @JmillsExpensify @trjExpensify how are we handling all of the money request/IOU changes?

For example, somethings that stand out to me:
- the top header part should basically just look like all other chat headers now. We also want to put the pay button up there.
- we removed the "To" field
- there would be just a wide Total line that sits directly above chat content

**Comment by shawnborton** [comment]
But if we're just breaking the changes out into chunks, I suppose this is fine?

**Comment by trjExpensify** [comment]
Here's [the main issue](https://github.com/Expensify/App/issues/20486) we were going from, I don't think this PR has come from it, and the changes to avatars and headers were being taken care of by @chiragsalian and @grgia. It does seem a bit unclear though, but I'll take that to the main issue and not in this PR.

**Comment by madmax330** [comment]
@trjExpensify so are we good to go with this PR or are we holding it for something?

**Comment by trjExpensify** [comment]
>@trjExpensify so are we good to go with this PR or are we holding it for something?

Yup, consensus in the main issue was that there isn't a conflict 👍

**Review by madmax330 — APPROVED** [review]

**Comment by Julesssss** [comment]
There was an outstanding lint issue here, maybe a new check was added recently?

**Comment by Julesssss** [comment]
It's likely we recently added/updated a check on main. So the issue wouldn't have been seen until the changes were merged with main (or main merged to the branch). Anyway, solve now 👍

**Comment by mvtglobally** [comment]
@ShogunFire @aimane-chnaif @madmax330 @AndrewGable No QA needed here?

**Comment by ShogunFire** [comment]
@mvtglobally Oups, looks like I forgot to add test steps sorry, I am doing this now

**Comment by ShogunFire** [comment]
@mvtglobally I have added the steps

**Comment by ShogunFire** [comment]
And I guess I should be follow the PR steps better next time because I just found a bug, when we are offline the scrollable header is not greyed out

**Comment by ShogunFire** [comment]
@aimane-chnaif @grgia Are you sure we can just ignore it ? It is now a deploy blocker :(

**Comment by madmax330** [comment]
@ShogunFire Go ahead and create your PR and we'll review

**Pull request merged**